lib/ansible/module_utils/csharp/Ansible.Basic.cs

```diff
@@ -83,6 +83,7 @@ public class AnsibleModule
             { "no_log", new List<object>() { false, typeof(bool) } },
             { "options", new List<object>() { typeof(Hashtable), typeof(Hashtable) } },
             { "removed_in_version", new List<object>() { null, typeof(string) } },
+            { "removed_at_date", new List<object>() { null, typeof(string) } },
```

Comment:
would be better as `typeof(DateTime)`

Reply:
@jborean93 what happens in the validate-modules test, when a argument spec containing `DateTime` objects is exported? Will they be converted properly?

Reply:
On PowerShell Core (which we use for validate-modules) it will be exported as a structured object that contains the ISO 8601 string and other values

```
PS /home/jborean> @{a = (Get-Date) } | ConvertTo-Json
{
  "a": {
    "value": "2020-03-27T07:02:17.147412+10:00",
    "DisplayHint": 2,
    "DateTime": "Friday, 27 March 2020 7:02:17 am"
  }
}
```

But even if we wanted to follow a similar structure to python or were unhappy with how the defaults look we can add our own custom formatting for a DateTime type here https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/ps_argspec.ps1#L35.

Reply:
Does the following look correct?
```
        } elseif ($value -is [DateTime]) {
            $Hash[$key] = $value.ToString("yyyy-MM-dd")
        }
```

Reply:
yep if you want it to be a string in that format that's how it would look.

Reply:
Yep. The validation should work the same for both Python and PowerShell modules.

One point though: if a different type than `DateTime` is used, `validate-module` will not detect that correctly. In particular, if a user uses `string` instead of `DateTime`, `validate-module` cannot detect that.

Reply:
That may be the case but IIRC the AnsibleModule parsing will fail to cast that type to a DateTime so the module wouldn't actually work.

Reply:
Except for `deprecatedAliases`, since there no check is done. I've added one in ed1da9be147048bf752e9900fe2bd515b687fbf1 (and hope it is correct).

Reply:
Ok, at least the tests do now pass again :)